src/request.ts

```diff
@@ -57,17 +66,24 @@ export default class Request {
           resolve(response);
         }).catch((e) => { throw(e); });
       });
+
       fetchPromise.catch(reject);
     });
   }
 
-  private buildHeaders(options: Object) : any {
+  private buildHeaders(options: FetchOptions) : any {
     let headers = {};
+
+    if (typeof options.headers == 'object') {
```

Comment:
But no actual model usage would cause this right? I've been treating `Request` as more or less private class (we could make this more explicit), I'm not sure we ever actually enter that situation in "real life". I just want to avoid trying to support things that don't really need support.